salt/utils/job.py

```diff
@@ -99,10 +99,10 @@ def store_job(opts, load, event=None, mminion=None):
         log.error(emsg)
         raise KeyError(emsg)
 
-    if 'jid' in load \
-            and 'get_load' in mminion.returners \
```

Comment:
@kunal-bajpai if I'm right you're trying to fix the error in this check where we have to use `and getfstr in minion.returners` to ensure it's there before calling, but we're checking for `'get_load'` which is never there.
If so please just fix this line, don't replace these checks with try. This will make your goal more clear.

Reply:
https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L102-L105
There is an issue with the above construct and i'll try and explain each one separately:-
1) https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L103
Not only does this check incorrectly look for `'get_load'` instead of `'{0}.get_load'.format(job_cache)`, this check is redundant as this is already checked for in the following construct, a few lines before this check appears
https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L93-L100
2) https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L104
This check is not required. It is intended to ensure that a document with key JID is not present in the job cache. However, the job of ensuring JIDs do not collide is of `prep_jid` which has been called before
https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L55-L57
Both the above checks can thus safely be removed
We are thus left with the following check only
```
if 'jid' in load:
    mminion.returners[savefstr](load['jid'], load)
```
3. Finally, the above construct tries to LBYL which is the opposite of the EAFP principle of Python.
An equivalent but more Pythonic construct is 
```
try:
        mminion.returners[savefstr](load['jid'], load)
 except KeyError as e:
        log.error("Load does not contain 'jid': %s", e)
```
Only `load['jid']` can generate a KeyError here as savefstr is also guaranteed to be present due to the earlier check
https://github.com/saltstack/salt/blob/b4ca024e44353bb9b5cc5e73d7de6db27a85ed80/salt/utils/job.py#L93-L100

@DmitryKuzmenko the goal of this PR is to fix the bug by cleaning up the problematic construct.

Reply:
@kunal-bajpai this makes sense. Thank you for explanation.